### PR TITLE
Initial implementation of triggerable text

### PR DESCRIPTION
**Pull request by joshuaskelly**
trigger-text

# Summary
Small feature that lets you set the text of Text entities via triggers.

### CODE REVIEW

File: Dungeoneer/src/com/interrupt/dungeoneer/entities/Text.java [hunk 1]
```diff
@@ -54,4 +53,13 @@ public void init(Level level, Level.Source source) {
             z += 0.4f; // Match offset to the new one of
         }
     }
+
+    @Override
+    public void onTrigger(Entity instigator, String value) {
+        if (value == null || value.equals("")) {
```
Comment (Interrupt) [inline Dungeoneer/src/com/interrupt/dungeoneer/entities/Text.java:59]: Make sure to call `super.onTrigger(instigator, value);` before this check, as trigger propagation happens in the base class there:
https://github.com/Interrupt/delverengine/blob/b0cd55a2258a4ee7d8cea9cbd910ead4afa33f8a/Dungeoneer/src/com/interrupt/dungeoneer/entities/triggers/BasicTrigger.java#L153

**Review by Interrupt — CHANGES_REQUESTED** [review]

File: Dungeoneer/src/com/interrupt/dungeoneer/entities/Text.java (hunk 1 above)
Comment (Interrupt) [reply Dungeoneer/src/com/interrupt/dungeoneer/entities/Text.java:59]: Nevermind that, the better method to wire up here would be to override `doTriggerEvent`, making sure to call the super of that at the top as well.

The difference here is that doTriggerEvent is called after the trigger delay, onTrigger before.

Comment (joshuaskelly) [reply Dungeoneer/src/com/interrupt/dungeoneer/entities/Text.java:59]: The `doTriggerEvent` only exists on `Trigger` objects.

Comment (Interrupt) [reply Dungeoneer/src/com/interrupt/dungeoneer/entities/Text.java:59]: Nevermind, this is fine as is then.

**Pull request merged**